sd/lb/round_robin.go

```diff
@@ -0,0 +1,39 @@
+package lb
+
+import (
+	"sync/atomic"
+
+	"github.com/go-kit/kit/sd"
+	"github.com/go-kit/kit/service"
+)
+
+// NewRoundRobin returns a load balancer that returns services in sequence.
+func NewRoundRobin(s sd.Subscriber) Balancer {
+	return &roundRobin{
+		s: s,
+		c: 0,
+	}
+}
+
+type roundRobin struct {
+	s sd.Subscriber
+	c uint64
+}
+
+func (rr *roundRobin) Service() (service.Service, error) {
+	services, err := rr.s.Services()
+	if err != nil {
+		return nil, err
+	}
+	if len(services) <= 0 {
+		return nil, ErrNoServices
+	}
+	var old uint64
+	for {
+		old = atomic.LoadUint64(&rr.c)
+		if atomic.CompareAndSwapUint64(&rr.c, old, old+1) {
```

Comment:
Maybe I'm misunderstanding this, but instead of doing a load+cas in a loop couldn't this just use an atomic add? For example:

```
old := atomic.AddUint64(&rr.c, 1) -1 // subtract 1 to get the previous value